echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/monitor/GitEventMonitor.java

```diff
@@ -94,7 +100,8 @@ protected boolean isSuccessfulTriggerEvent(TriggerEvent event) {
       && trigger.getSource().equalsIgnoreCase(source)
       && trigger.getProject().equalsIgnoreCase(project)
       && trigger.getSlug().equalsIgnoreCase(slug)
-      && (trigger.getBranch() == null || trigger.getBranch().equals("") || matchesPattern(branch, trigger.getBranch()));
+      && (trigger.getBranch() == null || trigger.getBranch().equals("") || matchesPattern(branch, trigger.getBranch()))
+      && hasValidGitHubSecureSignature(event, trigger);
```

Comment:
This should just apply to github triggers. I believe this is also used by Stash triggers